File: 98-validate-binary-search-tree/98-validate-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isValidBST(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        
        inorder = []
        
        def traversal(root):
            if root is None:
                return
            
            traversal(root.left)
            inorder.append(root.val)
            traversal(root.right)
            
        traversal(root)
        
        for i in range(0,len(inorder)-1):
            if inorder[i] >= inorder[i+1]:
                return False
            
        return True

        

File: 98-validate-binary-search-tree/98-validate-binary-search-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

void traversal(struct TreeNode* root, int* idx, int *result) {
    if (root == NULL) {
        return;
    }
    
    traversal(root->left, idx, result);
    result[(*idx)++] = root->val;
    traversal(root->right,idx, result);
}


bool isValidBST(struct TreeNode* root){
    if (root == NULL) {
        return true;
    }
    int idx = 0;
    int result[10000] = {0};
 
    traversal(root, &idx, result);
    
    int i;
    for (i = 0; i < idx-1; i++) {
        if (result[i] >= result[i+1]) {
            return false;
        }
    }

    return true;
}
    
    